packages/table/src/CellDefinitions/DateCell.tsx

```diff
@@ -1,20 +1,28 @@
 import React from 'react';
 import moment from 'moment';
+import { UncontrolledTooltip, UncontrolledTooltipProps } from 'reactstrap';
+import { Column, IdType, Row } from '../types/ReactTable';
 
-export type CellProps = {
+type CellProps<T extends IdType> = {
   value: string;
+  row: Row<T>;
+  column: Column<T>;
 };
 
-export interface DateTimeCellConfig {
+export interface DateTimeCellConfig<T> {
   dateFormat: string;
   defaultValue?: string | React.ReactChild | React.ElementType;
+  displayTooltip?: boolean;
+  tooltipProps?: UncontrolledTooltipProps;
 }
 
-const DateCell = ({
+const DateCell = <T extends IdType>({
   dateFormat,
   defaultValue,
-}: DateTimeCellConfig): JSX.Element | ((cell: CellProps) => JSX.Element) => {
-  const DateCellDef = ({ value }: CellProps): JSX.Element => {
+  displayTooltip = true,
+  tooltipProps,
+}: DateTimeCellConfig<T>): JSX.Element | ((cell: CellProps<T>) => JSX.Element) => {
+  const DateCellDef = ({ value, row, column }: CellProps<T>): JSX.Element => {
     let formattedValue;
```

Comment:
**SUGGESTION:** I know this isn't part of this change but while we are in here we could update this to ternary and join the declaration with the assignment. `const formattedValue = !value ? defaultValue : moment(value).format(dateFormat);`